spec/services/attribute_asserter_spec.rb

```diff
@@ -105,15 +103,92 @@
         end
 
         it 'silently skips invalid attribute name' do
-          expect(user.asserted_attributes).to have_key :email
-          expect(user.asserted_attributes).to_not have_key :foo
+          expect(user.asserted_attributes.keys).to eq([:uuid, :email])
         end
       end
     end
 
-    context 'un-verified user' do
-      let(:subject) { described_class.new(loa1_user, service_provider, authn_request) }
+    context 'verified user and LOA1 request' do
+      let(:subject) { described_class.new(user, service_provider, loa1_authn_request) }
+
+      context 'custom bundle includes email, phone, and first_name' do
+        before do
+          user.identities << identity
+          allow(service_provider.metadata).to receive(:[]).with(:attribute_bundle).
+            and_return(%w(email phone first_name))
+          subject.build
+        end
+
+        it 'only includes LOA1 attributes + uuid' do
+          expect(user.asserted_attributes.keys).to eq [:uuid, :email, :phone]
+        end
 
+        it 'does not create a getter function for LOA1 attributes' do
+          expect(user.asserted_attributes[:email][:getter]).to eq :email
+          expect(user.asserted_attributes[:phone][:getter]).to eq :phone
+        end
+
+        it 'gets UUID (MBUN) from Service Provider' do
+          uuid_getter = user.asserted_attributes[:uuid][:getter]
+          expect(uuid_getter.call(user)).to eq user.last_identity.uuid
+        end
+      end
+
+      context 'Service Provider does not specify bundle' do
+        before do
+          allow(service_provider.metadata).to receive(:[]).with(:attribute_bundle).
+            and_return(nil)
+          subject.build
+        end
+
+        context 'authn request does not specify bundle' do
+          it 'only returns uuid + phone' do
+            expect(user.asserted_attributes.keys).to eq [:uuid, :phone]
```

Comment:
I'm not sure if this is what is expected during LOA1 request. If SP does not request any attributes, we are sending back all Profile attributes by default for LOA3. Should there be a default set of attributes for LOA1 as well if the SP does not request any?

Should we always require the SP to request attributes and not send anything by default?